lib/Backends/NNPI/NNPIOptions.cpp

```diff
@@ -0,0 +1,437 @@
+  auto it = parameters->find(param);
+  if (it != parameters->end()) {
+    return it->second;
+  }
+  return defaultValue;
+}
+
+static llvm::StringMap<std::string> buildFilteredSupportedDescription(
+    std::vector<NNPIParamOption> filter,
+    const std::vector<std::vector<std::string>> &options) {
+  llvm::StringMap<std::string> map;
+  for (auto option : filter) {
+    std::stringstream desc;
+    desc << options[option][NNPIParamOptionDescription]
+         << " Default=" << options[option][NNPIParamOptionDefault]
+         << " Environment Variable:"
+         << options[option][NNPIParamOptionEnvironmentVarName];
+    map[options[option][NNPIParamOptionName]] = desc.str();
+  }
+  return map;
+}
+
+static std::vector<std::vector<std::string>> buildOptions() {
+  std::vector<std::vector<std::string>> options;
+  options.resize(NNPIParamOptionsSize);
+  options[UseIceTOption] = {"UseIceT", "USE_ICE_T",
+                            "Compile to ICE-T if true, ICE-Ref otherwise.",
+                            "0"};
+  options[InferOnDeviceOption] = {
+      "InferOnDevice", "USE_INF_API",
+      "Enable execution on device (UseIceT and !InferOnDevice will compile but "
+      "not execute inference).",
+      "0"};
+  options[InternalTestingOption] = {"[InternalTesting", "INVOKE_RUNTIME",
+                                    "Enable internal testing", ""};
+  options[CompiledFileOption] = {"CompiledFile", "ICE_T_FILE",
+                                 "Sets a file name to save the compilation "
+                                 "output to the filename specified.",
+                                 ""};
+  options[UseSymlowpOption] = {
+      "UseSymlowp", "SYMLOWP_WA",
+      "When this flag is set to true all quantized Int8 tensors are set to "
+      "Symlowp when their offset is 0.",
+      "0"};
+  options[DeviceVersionOption] = {"NNPIDeviceVersion", "NNPI_DEVICE_VERSION",
+                                  "Override target device version used for "
+                                  "compilation (currently supporting 1-3).",
+                                  "-1"};
+  options[NumOfWorkersOption] = {"NumOfWorkers", "NNPI_NUM_WORKERS",
+                                 "Override the amount of worker threads "
+                                 "allocated per network on the device.",
+                                 "-1"};
+  options[DeviceIDOption] = {
+      "DeviceID", "NNPI_DEVICE_ID",
+      "Override the target device ID used to run (0,1,...).", "-1"};
+  options[IceCoresOption] = {
+      "IceCores", "NNPI_ICE_CORES",
+      "Force compilation with specified amount of ice cores (1-12).", "-1"};
+  options[DeviceTraceOption] = {
+      "DeviceTracing", "NNPI_DEVICE_TRACING",
+      "Enabled device tracing (host2device, device2host copy infer etc.).",
+      "0"};
+  options[CompilationLogLevelOption] = {
+      "CompilationLogLevel", "NNPI_LOG_LEVEL",
+      "Sets the compilation logging level (0 most detailed).", "0"};
```

Comment:
nit: Perhaps mention what the max value accepted here is? Or even mention names of each level.